DuckDuckGo/NetworkProtectionRootView.swift

```diff
@@ -21,29 +21,43 @@
 
 import SwiftUI
 import NetworkProtection
+import Subscription
 
 @available(iOS 15, *)
 struct NetworkProtectionRootView: View {
-    let model = NetworkProtectionRootViewModel()
+    
+    let model = NetworkProtectionRootViewModel(featureActivation: AppDependencyProvider.shared.networkProtectionKeychainTokenStore)
+    let inviteViewModel: NetworkProtectionInviteViewModel
+    let statusViewModel: NetworkProtectionStatusViewModel
     let inviteCompletion: () -> Void
 
+    init(inviteCompletion: @escaping () -> Void) {
+        self.inviteCompletion = inviteCompletion
+        let accountManager = AppDependencyProvider.shared.subscriptionManager.accountManager
+        let redemptionCoordinator = NetworkProtectionCodeRedemptionCoordinator(isManualCodeRedemptionFlow: true,
+                                                                               accountManager: accountManager)
+        inviteViewModel = NetworkProtectionInviteViewModel(redemptionCoordinator: redemptionCoordinator, completion: inviteCompletion)
+        let locationListRepository = NetworkProtectionLocationListCompositeRepository(accountManager: accountManager)
+        statusViewModel = NetworkProtectionStatusViewModel(tunnelController: AppDependencyProvider.shared.networkProtectionTunnelController,
+                                                           settings: AppDependencyProvider.shared.vpnSettings,
+                                                           statusObserver: AppDependencyProvider.shared.connectionObserver,
+                                                           locationListRepository: locationListRepository)
+        // Prefetching this now for snappy load times on the locations screens
+        Task {
+            try? await locationListRepository.fetchLocationList()
+        }
```

Comment:
I moved this there because this call was previously in a sub-model init, it's never a good idea to add hidden and unrelated functionalities to an init, Imagine trying to test that model or trying to mock it, how do you mock the network call if it's hidden in an init?